frame/democracy/src/lib.rs

```diff
@@ -369,74 +376,96 @@ pub mod pallet {
 		/// The maximum number of public proposals that can exist at any time.
 		#[pallet::constant]
 		type MaxProposals: Get<u32>;
+
+		/// Ensure protocol root.
+		/// In case of substrate chain runtime it should same as `ensure_root`.
+		/// In case of some protocol instance, what ever check reasonble.
+		type EnsureRoot: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+
+		/// Runtime unique identifier for locking currency.
+		/// May be equivalent to PalletId.
+		#[pallet::constant]
+		type DemocracyId: Get<LockIdentifier>;
+
+		/// An origin on behalf of which the preimage calls are dispatched.
+		/// You may consider set it to be same as `Self::EnsureRoot`.
+		/// May be runtime wide or per set of pallets.
+		type ProtocolRoot: Get<<Self as Config<I>>::Origin>;
```

Comment:
Why are these types added, if the purpose of the PR is to only make the pallet instanciable?

Reply:
Purpose of pallet to make it meaningfully instantiable. By meaning I mean that second democracy should be able to execute on behalf of non system::Orign::Root (if 2 democracies execute as same origin, then why there are 2)? So you can execute via this on behalf of some account, for example anonymous proxy account without delegation or account derived from pallet id. So can do relay(parachain(protocol)) each has its own ""root"" and democracy. 

Same for currency, if we vote with same currency, why there is another democracy? So protocol can his own root and own  currency and own democracy.